pages/api/paybutton/download/transactions/[paybuttonId].ts

```
@@ -112,9 +43,15 @@ export default async (req: any, res: any): Promise<void> => {
     const userReqTimezone = req.headers.timezone as string
     const userPreferredTimezone = user?.preferredTimezone
     const timezone = userPreferredTimezone !== '' ? userPreferredTimezone : userReqTimezone
-
+    let networkIdArray = Object.values(NETWORK_IDS)
+    if (networkTicker !== undefined) {
+      const slug = Object.keys(NETWORK_TICKERS).find(key => NETWORK_TICKERS[key] === networkTicker)
+      const networkId = getNetworkIdFromSlug(slug ?? NETWORK_TICKERS.ecash)
```

Why are you defaulting to  `ecash`? Maybe if slug is invalid, we should just throw an error, we already have `INVALID_NETWORK_SLUG_400`.

Also could have `const slug =  NETWORK_SLUGS_FROM_IDS[NETWORK_IDS[ticker]]`

---

Also, this same codeblock appears right above `downloadTxsFile`, on both the only two places this function is ever called. Why not put it inside it, then?

---

> Also, this same codeblock appears right above `downloadTxsFile`, on both the only two places this function is ever called. Why not put it inside it, then?

Because the networkIdArray is used to fetch the txs something we are doing outside downloadTxsFile function